cdap-watchdog/src/main/java/io/cdap/cdap/metrics/process/DefaultMetricsWriterContext.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.metrics.process;
+
+import io.cdap.cdap.api.metrics.MetricsContext;
+import io.cdap.cdap.api.metrics.MetricsWriterContext;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Default implementation of {@link MetricsWriterContext}.
+ */
+public final class DefaultMetricsWriterContext implements MetricsWriterContext {
+
+  private static final String metricsFilePathKey = "metricsFilePath";
+  private final String metricsWriterFilePath;
+  private final MetricsContext metricsContext;
+
+  public DefaultMetricsWriterContext(MetricsContext metricsContext, String metricsWriterFilePath) {
+    this.metricsWriterFilePath = metricsWriterFilePath;
+    this.metricsContext = metricsContext;
+  }
+
+  public MetricsContext getMetricsContext() {
+    return metricsContext;
+  }
+
+  public Map<String, String> getProperties() {
+    Map<String, String> properties = new HashMap<>();
+    properties.put(metricsFilePathKey, metricsWriterFilePath);
```

Comment:
Why have this??

Reply:
The set of properties for a given metrics writer should be configurable and coming from CConfiguration

Reply:
fixed